discord/message.py

```diff
@@ -452,6 +455,87 @@ def guild_id(self) -> Optional[int]:
         return self._parent.guild_id
 
 
+class MessageSnapshot:
+    """Represents a message snapshot attached to a forwarded message.
+
+    Attributes
+    -----------
+    type: :class:`MessageType`
+        The type of the snapshotted message.
+    content: :class:`str`
+        The actual contents of the snapshotted message.
+    embeds: List[:class:`Embed`]
+        A list of embeds the snapshotted message has.
+    attachments: List[:class:`Attachment`]
+        A list of attachments given to the snapshotted message.
+    timestamp: :class:`datetime.datetime`
+        The snapshotted message's timestamp.
+    flags: :class:`MessageFlags`
+        Extra features of the the message snapshot.
+    """
+
+    __slots__ = (
+        '_cs_raw_channel_mentions',
+        '_cs_raw_mentions',
+        '_cs_raw_role_mentions',
+        '_edited_timestamp',
+        'attachments',
+        'content',
+        'embeds',
+        'flags',
+        'timestamp',
+        'type',
+    )
+
+    @classmethod
+    def _from_value(
+        cls,
+        state: ConnectionState,
+        message_snapshots: Optional[List[Dict[Literal['message'], MessageSnapshotPayload]]],
+    ):
+        if not message_snapshots:
+            return None
+
+        return [MessageSnapshot(state, snapshot['message']) for snapshot in message_snapshots]
+
+    def __init__(self, state: ConnectionState, data: MessageSnapshotPayload):
+        self.type: MessageType = try_enum(MessageType, data['type'])
+        self.content: str = data['content']
+        self.embeds: List[Embed] = [Embed.from_dict(a) for a in data['embeds']]
+        self.attachments: List[Attachment] = [Attachment(data=a, state=state) for a in data['attachments']]
+        self.timestamp: datetime.datetime = utils.parse_time(data['timestamp'])
+        self._edited_timestamp: Optional[datetime.datetime] = utils.parse_time(data['edited_timestamp'])
```

Comment:
```suggestion
        self._edited_at: Optional[datetime.datetime] = utils.parse_time(data['edited_timestamp'])
```

Same as above, but also.. we store this private attribute but don't seem to do anything with it, is this intended? Or would this be utilised as a property that's been missed?

Reply:
No, you're right, it should be a property

Reply:
Made a property, but I kept its name as _edited_timestamp since that's the same name as the private property in the Message class